cmd/minishift/cmd/oc_env.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/minishift/minishift/cmd/minishift/cmd/util"
+	instanceState "github.com/minishift/minishift/pkg/minishift/config"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/minishift/minishift/pkg/util/shell"
+	"github.com/spf13/cobra"
+	"os"
+	"path"
+	"text/template"
+)
+
+const (
+	ocEnvTmpl = `{{ .Prefix }}PATH{{ .Delimiter }}{{ .OcPath }}{{ .Suffix }}{{ .UsageHint }}`
+)
+
+type OcShellConfig struct {
+	shell.ShellConfig
+	OcPath    string
+	UsageHint string
+}
+
+func getOcShellConfig(api libmachine.API, forceShell string) (*OcShellConfig, error) {
```

Comment:
If we pass _instanceState.InstanceConfig.OcPath_ as string into this method, we can easily write some tests asserting that OCShellConfig contains what we want.